clients/fossid-webapp/src/test/kotlin/FossIdReportTest.kt

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (C) 2022 Bosch.IO GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+import com.github.tomakehurst.wiremock.WireMockServer
+import com.github.tomakehurst.wiremock.core.WireMockConfiguration
+
+import io.kotest.core.spec.style.StringSpec
+import io.kotest.matchers.collections.shouldContain
+import io.kotest.matchers.file.shouldHaveName
+import io.kotest.matchers.result.shouldBeSuccess
+import io.kotest.matchers.string.shouldEndWith
+
+import java.io.File
+
+import kotlin.io.path.createTempDirectory
+
+import org.ossreviewtoolkit.clients.fossid.FossIdRestService
+import org.ossreviewtoolkit.clients.fossid.generateReport
+import org.ossreviewtoolkit.clients.fossid.model.report.ReportType
+import org.ossreviewtoolkit.clients.fossid.model.report.SelectionType
+
+private const val SCAN_CODE = "semver4j_semver4j__20220119_094101_delta"
+
+class FossIdReportTest : StringSpec({
+    val server = WireMockServer(
+        WireMockConfiguration.options()
+            .dynamicPort()
+            .usingFilesUnderDirectory("src/test/assets/report")
+    )
+    lateinit var service: FossIdRestService
+    lateinit var directory: File
+
+    beforeSpec {
+        server.start()
+        service = FossIdRestService.create("http://localhost:${server.port()}")
+
+        directory = createTempDirectory("fossid_report").toFile()
+            .apply { deleteOnExit() }
+    }
+
+    afterSpec {
+        server.stop()
+    }
+
+    beforeTest {
+        server.resetAll()
+    }
+
+    "A dynamic HTML report can be generated for a scan" {
+        service.generateReport("", "", SCAN_CODE, ReportType.DYNAMIC, SelectionType.INCLUDE_FOSS, directory)
+            .let { result ->
+                result.shouldBeSuccess {
+                    it shouldHaveName
+                       "fossid-semver4j_semver4j__20220119_094101_delta-cc1267688905d7493df292786a245297c9fd36ee.html"
+                    val lines = it.readLines()
+                    lines shouldContain "<html>"
```

Comment:
Do we really need to read all lines here? How about something like `file.useLines { it shouldContain "<html>" }` instead, which uses a sequence?